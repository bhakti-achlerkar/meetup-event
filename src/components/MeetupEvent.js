import React, { Component } from 'react';
// import PropTypes from 'prop-types';
// import { connect } from 'react-redux';
import Attendees from './Attendees';
import queryString from 'query-string';
import ReactHtmlParser from 'react-html-parser';
import EventHeader from './EventHeader';

import {
  URL_NAME,
  MEETUP_API_BASE_URL
} from '../constants';

const initialState = {
  meetUpEvent: {
    name: '',
    venue: {
      name: ''
    },
    group: {
      name: ''
    }
  }
};

class MeetUpEvent extends Component {
  constructor(props) {
    super(props);
    this.state = {
      ...initialState
    };
  }

  componentDidMount() {
    const queryParams = queryString.stringify({
      'sign': true,
      'photo-host': 'public',
      'page': 1,
      'has_ended': false,
      'status': 'upcoming'
    });

    fetch(`${MEETUP_API_BASE_URL}${URL_NAME}/events?${queryParams}`, {
      headers: {
        'Content-type': 'application/json',
        'Accept': 'application/json'
      }
    })
      .then(response => response.json())
      .then(response => {
        this.setState({
          meetUpEvent: response[0]
        });
      }).catch(errObj => {
        console.log(errObj);
        this.setState({
          ...initialState
        });
      });
  }

  render() {
    const { meetUpEvent } = this.state;

    if(!meetUpEvent.name) return null;

    return (
      <div className='event-details'>
        <EventHeader
          duration={meetUpEvent.duration}
          eventName={meetUpEvent.name}
          groupName={meetUpEvent.group.name}
          eventDate={meetUpEvent.time}
          eventTime={meetUpEvent.local_time}
          venue={meetUpEvent.venue}></EventHeader>

        <div className="event-body">
          <h3>Description</h3>
          <div>{ReactHtmlParser(meetUpEvent.description)} </div>
        </div>
        <Attendees
          eventId={meetUpEvent.id}
          eventGroupName={URL_NAME}></Attendees>

      </div>
    );
  }
}

export default MeetUpEvent;
