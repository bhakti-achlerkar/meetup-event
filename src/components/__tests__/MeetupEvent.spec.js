import React from 'react';
import MeetupEvent from '../MeetupEvent';
import renderer from 'react-test-renderer';

const mockResponse = (status, statusText, response) => {
  return new Promise((resolve, reject) => {
    if(status == 200) {
      return resolve(response);
    } else {
      reject();
    }
  });};

window.fetch = jest.fn().mockImplementation(() =>
  Promise.resolve(mockResponse(200, null, "{name: '',venue:{name:''},group:{name:''}}")));

test('MeetupEvent rendered with event details', () => {
  const component = renderer.create(
    <MeetupEvent></MeetupEvent>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
