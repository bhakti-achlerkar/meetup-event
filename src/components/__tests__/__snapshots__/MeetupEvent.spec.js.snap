// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MeetupEvent rendered with event details 1`] = `null`;
